Core/Chat/TwitchChat.cs

```diff
@@ -0,0 +1,177 @@
+using TwitchLib.Communication.Clients;
+using TwitchLib.Communication.Models;
+
+namespace Core.Chat
+{
+    public sealed class TwitchChat : IChat
+    {
+        public event EventHandler<MessageEventArgs> IncomingMessage = null!;
+
+        private readonly ILogger<TwitchChat> _logger;
+        private readonly IClock _clock;
+        private readonly string _ircChannel;
+        private readonly ImmutableHashSet<IrcConfig.SuppressionType> _suppressions;
+        private readonly ImmutableHashSet<string> _suppressionOverrides;
+        private readonly IUserRepo _userRepo;
+        private readonly TwitchClient _twitchClient;
+
+        private bool _connected = false;
+
+        public TwitchChat(
+            ILoggerFactory loggerFactory,
+            IClock clock,
+            IrcConfig ircConfig,
+            IUserRepo userRepo)
+        {
+            _logger = loggerFactory.CreateLogger<TwitchChat>();
+            _clock = clock;
+            _ircChannel = ircConfig.Channel;
+            _suppressions = ircConfig.Suppressions;
+            _suppressionOverrides = ircConfig.SuppressionOverrides
+                .Select(s => s.ToLowerInvariant()).ToImmutableHashSet();
+            _userRepo = userRepo;
+
+            _twitchClient = new TwitchClient(
+                client: new TcpClient(new ClientOptions()),
+                logger: loggerFactory.CreateLogger<TwitchClient>());
+            var credentials = new ConnectionCredentials(
+                twitchUsername: ircConfig.Username,
+                twitchOAuth: ircConfig.Password);
+            _twitchClient.Initialize(
+                credentials: credentials,
+                channel: ircConfig.Channel,
+                // disable TwitchLib's command features, we do that ourselves
+                chatCommandIdentifier: '\0',
+                whisperCommandIdentifier: '\0');
+        }
+
+        public Task SendMessage(string message)
+        {
+            if (_suppressions.Contains(IrcConfig.SuppressionType.Message) &&
+                !_suppressionOverrides.Contains(_ircChannel))
+            {
+                _logger.LogDebug($"(suppressed) >#{_ircChannel}: {message}");
+                return Task.CompletedTask;
+            }
+            _logger.LogDebug($">#{_ircChannel}: {message}");
+            _twitchClient.SendMessage(_ircChannel, message);
+            return Task.CompletedTask;
+        }
+
+        public Task SendWhisper(User target, string message)
+        {
+            if (_suppressions.Contains(IrcConfig.SuppressionType.Whisper) &&
+                !_suppressionOverrides.Contains(target.SimpleName))
+            {
+                _logger.LogDebug($"(suppressed) >@{target.SimpleName}: {message}");
+                return Task.CompletedTask;
+            }
+            _logger.LogDebug($">@{target.SimpleName}: {message}");
+            _twitchClient.SendWhisper(target.SimpleName, message);
+            return Task.CompletedTask;
```

Comment:
TwitchLib doesn't do async, but the interface felt like it was supposed to be async anyway. That's why it's just returning `Task.CompletedTask` in case you wondered

Reply:
do we want to make these call Task.Run instead?